src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
```

Comment:
That's an interesting point, and quite probably right - I'm happy to not use enumerables, and just do the set of null checks if we're concerned? (The last commit is the refactoring, but shouldn't functionally change things)

I can try and do some benchmarks, not sure how much this will tell us though?